drivers/display/display_gc9x01x.c

```diff
@@ -0,0 +1,691 @@
+			 0x70U, 0x70U},
+	},
+	{
+		.cmd = 0x64U,
+		.len = 7U,
+		.data = {0x28U, 0x29U, 0xF1U, 0x01U, 0xF1U, 0x00U, 0x07U},
+	},
+	{
+		.cmd = 0x66U,
+		.len = 10U,
+		.data = {0x3CU, 0x00U, 0xCDU, 0x67U, 0x45U, 0x45U, 0x10U, 0x00U, 0x00U, 0x00U},
+	},
+	{
+		.cmd = 0x67U,
+		.len = 10U,
+		.data = {0x00U, 0x3CU, 0x00U, 0x00U, 0x00U, 0x01U, 0x54U, 0x10U, 0x32U, 0x98U},
+	},
+	{
+		.cmd = 0x74U,
+		.len = 7U,
+		.data = {0x10U, 0x85U, 0x80U, 0x00U, 0x00U, 0x4EU, 0x00U},
+	},
+	{
+		.cmd = 0x98U,
+		.len = 2U,
+		.data = {0x3EU, 0x07U},
+	},
+};
+
+static int gc9x01x_transmit(const struct device *dev, uint8_t cmd, const void *tx_data,
+			    size_t tx_len)
+{
+	const struct gc9x01x_config *config = dev->config;
+	int retval;
+	struct spi_buf tx_buf = {.buf = &cmd, .len = 1U};
+	struct spi_buf_set tx_bufs = {.buffers = &tx_buf, .count = 1U};
+
+	retval = gpio_pin_set_dt(&config->cmd_data, GC9X01X_GPIO_LEVEL_CMD);
+	if (retval < 0) {
+		return retval;
+	}
+	retval = spi_write_dt(&config->spi, &tx_bufs);
+	if (retval < 0) {
+		return retval;
+	}
+
+	/* send data (if any) */
+	if (tx_data != NULL) {
+		tx_buf.buf = (void *)tx_data;
+		tx_buf.len = tx_len;
+
+		retval = gpio_pin_set_dt(&config->cmd_data, GC9X01X_GPIO_LEVEL_DATA);
+		if (retval < 0) {
+			return retval;
+		}
+		retval = spi_write_dt(&config->spi, &tx_bufs);
+		if (retval < 0) {
+			return retval;
+		}
+	}
+
+	return 0;
+}
+
+static int gc9x01x_regs_init(const struct device *dev)
+{
+	const struct gc9x01x_config *config = dev->config;
+	const struct gc9x01x_regs *regs = config->regs;
+	int retval;
+
+	/* Enable inter-command mode */
+	retval = gc9x01x_transmit(dev, GC9X01X_CMD_INREGEN1, NULL, 0);
+	retval |= gc9x01x_transmit(dev, GC9X01X_CMD_INREGEN2, NULL, 0);
+
+	/* Apply default init sequence */
+	for (int i = 0; (i < ARRAY_SIZE(default_init_regs)) && (retval == 0); i++) {
+		retval |= gc9x01x_transmit(dev, default_init_regs[i].cmd, default_init_regs[i].data,
+					   default_init_regs[i].len);
+	}
+
+	/* Apply generic configuration */
+	retval |=
+		gc9x01x_transmit(dev, GC9X01X_CMD_PWRCTRL2, regs->pwrctrl2, sizeof(regs->pwrctrl2));
+	retval |=
+		gc9x01x_transmit(dev, GC9X01X_CMD_PWRCTRL3, regs->pwrctrl3, sizeof(regs->pwrctrl3));
+	retval |=
+		gc9x01x_transmit(dev, GC9X01X_CMD_PWRCTRL4, regs->pwrctrl4, sizeof(regs->pwrctrl4));
+	retval |= gc9x01x_transmit(dev, GC9X01X_CMD_GAMMA1, regs->gamma1, sizeof(regs->gamma1));
+	retval |= gc9x01x_transmit(dev, GC9X01X_CMD_GAMMA2, regs->gamma2, sizeof(regs->gamma2));
+	retval |= gc9x01x_transmit(dev, GC9X01X_CMD_GAMMA3, regs->gamma3, sizeof(regs->gamma3));
+	retval |= gc9x01x_transmit(dev, GC9X01X_CMD_GAMMA4, regs->gamma4, sizeof(regs->gamma4));
+	retval |= gc9x01x_transmit(dev, GC9X01X_CMD_FRAMERATE, regs->framerate,
+				   sizeof(regs->framerate));
+
+	/* Enable Tearing line */
+	retval |= gc9x01x_transmit(dev, GC9X01X_CMD_TEON, NULL, 0);
+
+	if (retval == 0) {
+		return 0;
+	} else {
+		return -EIO;
+	}
```

Comment:
see my comment above, return on the individual errors, `return 0;` here

Reply:
Fixed above [a1256a9](https://github.com/zephyrproject-rtos/zephyr/commit/a1256a9156de7aca7ae9d46e4335310a8bb595bb)